mongo/integration/causal_consistency_test.go

```diff
@@ -193,21 +192,6 @@ func TestCausalConsistency_Supported(t *testing.T) {
 		assert.NotNil(mt, sentOptime, "expected operation time on command, got nil")
 		assert.True(mt, currOptime.Equal(*sentOptime), "expected operation time %v, got %v", currOptime, sentOptime)
 	})
-	unackWcOpts := options.Collection().SetWriteConcern(writeconcern.New(writeconcern.W(0)))
-	mt.RunOpts("unacknowledged write", mtest.NewOptions().CollectionOptions(unackWcOpts), func(mt *mtest.T) {
```

Comment:
Just curious, in what situations does this test fail? It does not fail locally for me against a 5.0 replica set.
> do we want to tweak the test to assert that doing an unacknowledged write within a session results in an error?

If this is client-side error (thrown within the driver), then that could be a valuable test. If the server just doesn't support unacknowledged writes with explicit sessions, then I don't think we need to test that behavior on our side.

Reply:
The test should fail if the `0`-number  write concern being sent over the wire is acknowledged, which will [request[s] no acknowledgment of the write operation](https://docs.mongodb.com/manual/reference/write-concern/#mongodb-writeconcern-writeconcern.-number-).

The error itself is thrown by the `*mongo.Collection.InsertOne` method, specifically whenever the wire message flag is `MoreToCome`.  I can see _[how](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/#flag-bits)_ we determine this message from [the code base](https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/wiremessage/wiremessage.go#L336), but have no idea how the relationship between unacknowledged writes and `moreToCome` is 1-1.  The server does not explicitly handle this, and it is [technically recommended behavior](https://github.com/mongodb/specifications/blame/master/source/sessions/driver-sessions.rst#L653).